app/code/Magento/Catalog/Controller/Adminhtml/Product/Save.php

```diff
@@ -117,6 +117,7 @@ public function execute()
                 $productId = $product->getEntityId();
                 $productAttributeSetId = $product->getAttributeSetId();
                 $productTypeId = $product->getTypeId();
+                $this->getRequest()->setParam('type', $productTypeId);
```

Comment:
and it appears only in the first save (add attribute for simple and not configurable product, set this attribute to be required. Then create new simple product, add children product, save).